routers/credential.py

```
@@ -0,0 +1,42 @@
+import json
+from fastapi import APIRouter, Response, status
+
+from schemas import SecuredCredentialWithOptions
+from schemas.mappings.vcr_credential import VCRCredential
+from services import vcr as vcr_service
+
+router = APIRouter(prefix="/credentials", tags=["credentials"])
+
+
+@router.post(
+    "/",
+    status_code=status.HTTP_201_CREATED,
+    response_model=SecuredCredentialWithOptions,
+    response_model_exclude_none=True,
+    response_model_by_alias=True,
+)
+async def issue_credential(secured_credential: SecuredCredentialWithOptions):
+    """Issue a new credential"""
+
+    try:
+        secured_credential_data = secured_credential.model_dump(
+            by_alias=True, exclude_none=True
+        )
+        vcr_credential = VCRCredential(**secured_credential_data)
+
+        data = vcr_credential.model_dump(by_alias=True, exclude_none=True)
+        svc_response = await vcr_service.issue_credential(data)
+        # TODO: Log the response
+        print(svc_response)
+    except Exception as e:
+        # TODO: Log the error
+        print(e)
+        return Response(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content=json.dumps(
+                {"error": "There was an error issuing the credential"}
+            ),
+            media_type="application/json",
+        )
```

I like to import the `HTTPException` class from fastapi and raise an error for error handling. You can also raise an error from a package that isn't directly returning a Response.

```python
from fastapi import HTTPException
[...]
raise HTTPException(status_code=400, detail="There was an error issuing the credential.")
```



Another tip I like to use is to return a `JSONResponse` object if its not an error.

```python
from fastapi.responses import JSONResponse
[...]
return JSONResponse(status_code=200, content={})
```